src/live_restore/live_restore_fs.c

```diff
@@ -355,8 +365,8 @@ __live_restore_fs_directory_list_worker(WT_FILE_SYSTEM *fs, WT_SESSION *wt_sessi
                  * We're iterating files in the source, but we want to check if they exist in the
                  * destination, so create the file path to the backing destination file.
                  */
-                WT_ERR(__live_restore_create_file_path(
-                  session, &lr_fs->destination, dirlist_src[i], &temp_path));
+                WT_ERR(
+                  __live_restore_create_file_path(session, directory, dirlist_src[i], &temp_path));
```

Comment:
Is there a guarantee that `directory` is always `destination.home`? It should be possible that it's a subfolder in the destination. I feel like there should be a failing catch2 test after this change.

Reply:
Ok, this does look like it is creating a pathname in the destination, so try to use `wt_filename_construct`.

Reply:
Fixed it here but had to modify the implementation of `wt_filename_construct` to work with loops. [e7994b1](https://github.com/wiredtiger/wiredtiger/pull/11453/commits/e7994b13c9d916f08dc7bfc3ff0ac9c0dc8effb7)

Reply:
> Is there a guarantee that directory is always destination.home? 
Would making this path_dest fix that?